src/db.rs

```diff
@@ -209,29 +209,106 @@ pub enum IteratorMode<'a> {
     From(&'a [u8], Direction),
 }
 
-impl<'a> DBRawIterator<'a> {
-    fn new(db: &DB, readopts: &ReadOptions) -> DBRawIterator<'a> {
-        unsafe {
-            DBRawIterator {
-                inner: ffi::rocksdb_create_iterator(db.inner, readopts.inner),
-                db: PhantomData,
-            }
-        }
+pub trait CreateIter {
```

Comment:
@hjiayz could you please describe why you add this crate ?

Reply:
make DBRawIterator and DBIterator support Transaction.